common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -25,6 +25,8 @@
 import com.google.common.collect.Lists;
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.util.DiskChecker;
```

Comment:
So if you aren't using yarn recovery, do you care about this file at all?  ie in your cluster if a nodemanager dies you don't expect it to come back up and be able to serve shuffle data?  
If not perhaps we should make it configurable to not store it in the levelDB at all.

I'd have to look to see when the recovery feature was added but as long as that is in early enough hadoop we can just make it dependent upon the yarn recovery setting.